### PR TITLE
Added workersList() method into queue

### PR DESCRIPTION
**Pull request by kolomiichenko**
`workersList()` - a function returning the array of items currently being processed. This can be useful when it is not need push the new job to the queue if it's in process.

### CODE REVIEW

**Comment by sadovnychyi** [comment]
[LGTM](http://lgtm.in/i/qJOa3EHRp)

[:+1:](http://lgtm.in/u/qJOa3EHRp)

**Comment by aearly** [comment]
Why do you need this?  What problem does it solve?

**Comment by kolomiichenko** [comment]
@aearly, in my situation, every task working for a few seconds(minutes), so I need to display progress queue. But in queue.tasks object I can't see tasks currently being processed. This code will solve this problem. Secondly, his can be useful when it is not need push the new job to the queue if it's in process.

File: lib/async.js [hunk 1]
```diff
@@ -863,8 +863,17 @@
         function _next(q, tasks) {
             return function(){
                 workers -= 1;
+
```
Comment (megawac) [inline lib/async.js]: spliced?

Comment (kolomiichenko) [inline lib/async.js]: @megawac, I do not know what I guided when giving named. Renamed to `removed`. :smile:

**Comment by megawac** [comment]
@aearly @kolomiichenko is this actually useful? I feel like you can get the same information on the progress by inspecting `queue.length`, `queue.running` and `queue.concurrency`

I want to better understand this use case, as this has become a bottleneck for queue

**Comment by aearly** [comment]
I feel like the above comment explains it pretty well.  Sometimes it is useful to know exactly which objects are being processed by the queue.

**Pull request merged**